frontend/src/components/RecommendedRecipes.js

```diff
@@ -120,6 +120,7 @@ class RecommendedRecipes extends Component {
 
   setRecipeAndRedirect(recipe) {
     localStorage.setItem("recipe", JSON.stringify(recipe));
```

Comment:
How expensive is the trip to localStorage? I think it's on the backend? Here it's the same machine and only involves disk access but when this project is scaled it might involve network roundtrip + disk access.

Instead of making three trips to the local storage here (and whenever you want to retrieve a recipe), consider changing the JSON itself to a different structure so it holds
recipe: {ingredients, full_recipe, tutorial-step}

Reply:
localStorage in this case is the data that is stored locally on the user's browser, so it doesn't go to our backend server. Could you please clarify how that would result in network roundtrip when the project is scaled?

This method only gets called when a user presses on "Let's Go" to proceed with a chosen recipe. Then the user gets redirected to CookRecipe.js which retrieves the recipe from localStorage only once when the page loads. All attributes (ingredients, instructions) are then accessed from the local state of CookRecipe component which means localStorage is not touched. On the other hand, tutorial step is modified on every change of the step, so I thought it would make sense to separate it.

Reply:
If it's on the user's browser, that's great. I thought we were storing this on the backend. (in which case it would've resulted in a network trip)

Few other considerations of using localStorage:
1. Nothing is sent back to your server, so if there's any state you'd like to persist across devices, you need to look into cookies or adding something to the user's data on the server.
2. If the user accesses the site from an Incognito session, localStorage is cleared at the end of the session and nothing is persisted across browser sessions.
3. There are per-origin limits to how much you can store in localStorage. Check you have quota left before storing to it or check for the quota_exceeded error. Read more here:
https://developer.chrome.com/apps/offline_storage
4. (Can be a follow-up CL): Implement a strategy for deleting older recipes if you'd prefer to overwrite localStorage with most recent recipes.

Reply:
Thank you for your suggestions! I will look into cookies to persist the state across devices and minimise the use of localStorage. That will be done in the follow-up CL.